source/rust_verify/example/power_of_2.rs

```diff
@@ -0,0 +1,183 @@
+// Some of the lemmas are ported from the following:
+// https://github.com/dafny-lang/libraries/blob/master/src/NonlinearArithmetic/Power2.dfy
+// https://github.com/dafny-lang/libraries/blob/master/src/NonlinearArithmetic/Power.dfy
+
+#[allow(unused_imports)]
+use builtin::*;
+#[allow(unused_imports)]
+mod pervasive;
+#[allow(unused_imports)]
+use pervasive::*;
+#[allow(unused_imports)]
+use builtin_macros::*;
+
+
+#[spec]
+fn pow2(e: nat) -> nat {
```

Comment:
Should this file use the `verus!` syntax?

Reply:
I think I encountered small problems around automatic widening(to `int`), but I think I should try again.